content-tools/update-cdids/main.go

```diff
@@ -0,0 +1,443 @@
+//		&overwriteExisting=true
+
+func addCDIDToCollection(ctx context.Context, client *http.Client, environment string, collectionID string, cdIDLocation string, data string) error {
+	cdIDURI := strings.Replace(cdIDLocation, environment, "", -1)
+
+	pageURL := fmt.Sprintf("%s/zebedee/content/%s?uri=%s/data.json&overwriteExisting=true", environment, collectionID, cdIDURI)
+	req, err := http.NewRequestWithContext(ctx, "POST", pageURL, bytes.NewBuffer([]byte(data)))
+	if err != nil {
+		errMessage := fmt.Errorf("failed to page adding to collection request. Error: %v", err)
+		log.Error(errMessage)
+		return errMessage
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set(ZebedeeToken, ctx.Value(ZebedeeToken).(string))
+
+	resp, err := client.Do(req)
+	if err != nil {
+		errMessage := fmt.Errorf("failed to add collection to page. Error: %v", err)
+		log.Error(errMessage)
+
+		return errMessage
+	}
+	if resp.StatusCode > 400 {
+		errMessage := fmt.Errorf("failed to add collection to page. Error In API: %v. Status Code: %s", err, resp.Status)
+		log.Error(errMessage)
+
+		return errMessage
+	}
+	return nil
+}
+
+func fetchDataForCDID(ctx context.Context, client *http.Client, environment string, location string) (string, error) {
+
+	cdIDDataURL := fmt.Sprintf("%s/data", location)
+
+	req, err := http.NewRequestWithContext(ctx, "GET", cdIDDataURL, nil)
+	if err != nil {
+		errMessage := fmt.Errorf("failed to prepare fetch CDID request. Error: %v", err)
+		log.Error(errMessage)
+		return "", errMessage
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		if err != nil {
+			errMessage := fmt.Errorf("failed to fetch CDID data from API. Error: %v", err)
+			log.Error(errMessage)
+
+			return "", errMessage
+		}
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		errMessage := fmt.Errorf("failed to read CDID data response. Error: %v", err)
+		log.Error(errMessage)
+		return "", errMessage
+	}
+
+	return string(body), nil
+}
+
+func searchOldCdID(ctx context.Context, client *http.Client, environment string, cdID string) (string, error) {
+	searchURL := fmt.Sprintf("%s/search?q=%s", environment, cdID)
+
+	req, err := http.NewRequestWithContext(ctx, "GET", searchURL, nil)
+	if err != nil {
+		errMessage := fmt.Errorf("failed to prepare search request. Error: %v", err)
+		log.Error(errMessage)
+		return "", errMessage
+	}
+	resp, err := client.Transport.RoundTrip(req)
+	if err != nil {
+		if err != nil {
+			errMessage := fmt.Errorf("failed to search API. Error: %v", err)
+			log.Error(errMessage)
+
+			return "", errMessage
+		}
+	}
+
+	defer resp.Body.Close()
+	locationHeader := resp.Header.Get("Location")
+	if len(locationHeader) == 0 {
+		errMessage := fmt.Errorf("failed to search the given CDID")
+		log.Error(errMessage)
+
+		return "", errMessage
+	}
+
+	return strings.Split(locationHeader, "?")[0], nil
+}
+
+type CdIDPair struct {
+	cdID    string
+	oldCdID string
+}
```

Comment:
Move this to the top of the file -> general go file structure convention is:
- Package doc
- Imports
- vars & const
- types
- funcs & methods.
Picky I know - but having the types all in one place makes the files easier to navigate and know where to find things.